src/components/otpBox/index.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+export const OTPBox = (props) => {
+    const {
+        OTPInputIds,
+        otp,
+        setOtp,
+        inputsRef
+    } = props;
+    const lengthOfOTP = OTPInputIds.length;
+
+    const handleOTPinput = (event) => {
+        const { value, dataset: { otpid, index } } = event.target;
+        if (otp[otpid] === '') {
+            setOtp({ ...otp, [otpid]: isOneDigitNumber(value) ? value : otp[otpid] });
```

Comment:
reassigning the existing state value again is not necessary here.
Instead, the isOneDigitNumber(value) can be checked before and state can be updated in that if case.
If it returns false, no need to update the state.